pkg/controllers/clusteroperator/reconcile_test.go

```diff
@@ -0,0 +1,165 @@
+package clusteroperator
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/client-go/config/clientset/versioned/fake"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	common "github.com/openshift/ibm-roks-toolkit/pkg/controllers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func TestReconcile(t *testing.T) {
+	versions := map[string]string{
+		"release":    "alpha",
+		"kubernetes": "beta",
+	}
+
+	testCases := []struct {
+		testCase    string
+		existingCOs []runtime.Object
+	}{
+		{
+			testCase: "when there's no cluster operators",
+		},
+		{
+			testCase: "when a clusterOperator exists with no status",
+			existingCOs: []runtime.Object{
+				&configv1.ClusterOperator{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "openshift-apiserver",
+					},
+				},
+			},
+		},
+		{
+			testCase: "when a clusterOperator exists with outdated versions",
+			existingCOs: []runtime.Object{
+				&configv1.ClusterOperator{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "openshift-apiserver",
+					},
+					Status: configv1.ClusterOperatorStatus{
+						Conditions: nil,
+						Versions: []configv1.OperandVersion{
+							{
+								Name:    "openshift-apiserver",
+								Version: "outdated",
+							},
+							{
+								Name:    "operator",
+								Version: "outdated",
+							},
+						},
+						RelatedObjects: nil,
+						Extension:      runtime.RawExtension{},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.testCase, func(t *testing.T) {
+			fakeClient := fake.NewSimpleClientset(tc.existingCOs...)
+			informerFactory := configinformers.NewSharedInformerFactory(fakeClient, common.DefaultResync)
+			stopCh := make(chan struct{})
+			go informerFactory.Start(stopCh)
+			go informerFactory.Config().V1().ClusterOperators().Informer().Run(stopCh)
```

Comment:
You only need to do the informerFactory.Start() on line 77, no need to separately start the clusteroperators informer. However before calling informerFactory.Start, you need to have accessed the the clusteroperator informer, so that it gets started.

I would change to something like:
```go
informer := informerFactory.Config().V1().ClusterOperators().Informer()
go informerFactory.Start(stopCh)
```

Also you don't need to do the PollImmediate below if you wait for the informer cache to be synced, so I would add
```go
cache.WaitForSync(stopCh, informer.HasSynced)
```
right after informerFactory.Start() where `cache` comes from `k8s.io/client-go/tools/cache`